core/src/main/java/io/questdb/griffin/engine/join/CrossJoinRecordCursorFactory.java

```diff
@@ -59,11 +59,18 @@ public void close() {
 
     @Override
     public RecordCursor getCursor(SqlExecutionContext executionContext) throws SqlException {
-        final RecordCursor masterCursor = masterFactory.getCursor(executionContext);
-        RecordCursor slaveCursor = slaveFactory.getCursor(executionContext);
-        if (masterCursor.hasNext()) {
-            cursor.of(masterCursor, slaveCursor);
-            return cursor;
+        RecordCursor masterCursor = masterFactory.getCursor(executionContext);
+        RecordCursor slaveCursor = null;
+        try {
+            slaveCursor = slaveFactory.getCursor(executionContext);
+            if (masterCursor.hasNext()) {
+                cursor.of(masterCursor, slaveCursor);
+                return cursor;
+            }
+        } catch (Throwable ex) {
+            Misc.free(masterCursor);
```

Comment:
I think we need a similar catch block in `getCursor` method in the following cursor factories:
* `AsOfJoinLightRecordCursorFactory`
* `AsOfJoinNoKeyRecordCursorFactory`
* `AsOfJoinRecordCursorFactory`
* `HashJoinLightRecordCursorFactory` (this one has a catch block, but the try block doesn't include the master factory's `getCursor` call)
* `HashJoinRecordCursorFactory` (same as above)
* `HashOuterJoinLightRecordCursorFactory` (same as above)
* `HashOuterJoinRecordCursorFactory` (same as above)
* `LtJoinLightRecordCursorFactory`
* `LtJoinNoKeyRecordCursorFactory`
* `LtJoinRecordCursorFactory`
* `SpliceJoinLightRecordCursorFactory`
* `UnionAllRecordCursorFactory`

Reply:
Good spot, fixed those too